topics/reference_guide/custom_language_support/additional_minor_features.md

```diff
@@ -121,6 +121,17 @@ EP: `com.intellij.include.provider`
 [`FileIncludeProvider`](upsource:///platform/lang-impl/src/com/intellij/psi/impl/include/FileIncludeProvider.java) provides information about _include_ statements resolving to files (e.g., `<xi:include>` in XML).
 Including/included files can then be obtained via [`FileIncludeManager`](upsource:///platform/lang-api/src/com/intellij/psi/impl/include/FileIncludeManager.java).
 
+### Recognizing Complex Multi-Block Expressions
+
+EP: `com.intellij.codeBlockSupportHandler`
+
+[`CodeBlockSupportHandler`](upsource:///platform/lang-impl/src/com/intellij/codeInsight/highlighting/CodeBlockSupportHandler.java)
+allows providing text ranges for more complex code blocks like, e.g., in `if`/`elsif`/`else` blocks.
+It is used to highlight markers and keywords if one is under the cursor, and for navigation to the beginning/end of blocks.
+See also the [paired brace matching](additional_minor_features.md#brace-matching).
```

Comment:
add reverse link there as well, can the link be shortened to just `(#header)`?

Reply:
Fixed and yes, you can just use `#header` it seems.